Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -138,7 +137,11 @@ def __dafnystr__(self) -> str:
         return '[' + ', '.join(map(string_of, self)) + ']'
 
     def __add__(self, other):
-        return Seq(super().__add__(other), isStr=self.isStr and other.isStr)
+        if self.isStr is None or other.isStr is None:
+            isStr = None
+        else:
+            isStr = self.isStr and other.isStr
+        return Seq(super().__add__(other), isStr=isStr)
```

Comment:
Are you sure that this changes the semantics?

Reply:
Yup. Before `Seq("a", isStr=None) + Seq("b", isStr=None)` would become the same as `Seq("ab", isStr=False)`, which would still go through the inference and decide it was a string, which we don't want.

Reply:
```
>>> (None and None) == None
True
>>> (None and None) == False
False
```

Reply:
Ah you're right, `self.isStr = None` is all that needed to change. I added this at the same time and never tested it without. :) Thanks!